src/network/providers/provider.ts

```diff
@@ -0,0 +1,141 @@
+};
+
+export const tldToChain: Record<string, ChainId> = {
+    '.sol': ChainId.SolanaMainnet,
+    '.eth': ChainId.EthereumRinkeby
+};
+
+// Type guard to determine if a given string|number is a valid ChainId
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function isChainId(chainId: any): chainId is ChainId {
+    return Object.values(ChainId).includes(chainId);
+}
+
+/**
+ * BlockchainProvider is the abstraction that consolidates and exposes
+ * blockchain functionality and provides a unified solution for working
+ * with different chains such as Ethereum and Solana.
+ */
+class BlockchainProvider implements Provider {
+    private providers: {[key in Blockchain]?: Provider} = {};
+
+    private getProvider(chainId: ChainId): Provider {
+        const blockchain = chainToBlockchain[chainId];
+        if (!this.providers[blockchain]) {
+            switch (blockchain) {
+                case Blockchain.ETHEREUM:
+                    this.providers[blockchain] = EthereumProvider.getInstance();
+                    break;
+                case Blockchain.SOLANA:
+                    this.providers[blockchain] = SolanaProvider.getInstance();
+                    break;
+                default:
+                    const errMsg = `Unsupported chain ID "${chainId}".`;
+                    log.error({chainId}, errMsg);
+                    throw new Error(errMsg);
+            }
+        }
+        return this.providers[blockchain] as Provider;
+    }
+
+    private notImplementedErr(chainId: ChainId, method: string): Error {
+        const errMsg = `Method ${method} not implemented by ${this.name(chainId)}`;
+        log.error({chainId, method}, errMsg);
+        return new Error(errMsg);
+    }
+
+    public name(chainId: ChainId): string {
+        return this.getProvider(chainId).name(chainId);
+    }
+
+    public async resolveDomain(chainId: ChainId, domain: string): Promise<DomainRegistry> {
+        const registry = await this.getProvider(chainId).resolveDomain(chainId, domain);
+        return registry;
+    }
+
+    public async setDomainContent(
+        chainId: ChainId,
+        domain: string,
+        data: string
+    ): Promise<Record<string, string|number|boolean>> {
+        const tx = await this.getProvider(chainId).setDomainContent(chainId, domain, data);
+        return tx;
+    }
+
+    public async send(chainId: ChainId, data: RPCRequest): Promise<JsonRpcResponse> {
+        const provider = this.getProvider(chainId);
+        const resp = await provider.send(chainId, data);
+        return resp;
+    }
+
+    public async sendTransaction(
+        chainId: ChainId,
+        reqId: number,
+        txParams: unknown[]
+    ): Promise<JsonRpcResponse> {
+        const resp = await this.getProvider(chainId).sendTransaction(chainId, reqId, txParams);
+        return resp;
+    }
+
+    public async requestAccounts(chainId: ChainId, reqId: number): Promise<JsonRpcResponse> {
```

Comment:
All these methods could be surrounded in try/catch, but I think it's better for the caller to handle errors. We are already have some duplication when it comes to handling and logging errors.